qa-tests-backend/src/main/groovy/util/CollectorUtil.groovy

```diff
@@ -0,0 +1,109 @@
+package util
+
+import static util.Helpers.withRetry
+
+import groovy.util.logging.Slf4j
+import groovy.transform.CompileStatic
+
+import orchestratormanager.Kubernetes
+
+import com.google.protobuf.util.JsonFormat
+
+import sensor.Collector
+
+@Slf4j
+@CompileStatic
+class CollectorUtil {
+    static final String RUNTIME_CONFIG_MAP_NAME = "collector-config"
+    static final String RUNTIME_CONFIG_MAP_KEY = "runtime_config.yaml"
+
+    static final String ENABLED_VALUE = "ENABLED"
+    static final String DISABLED_VALUE = "DISABLED"
+
+    static parseJsonToProtobuf(String json) {
+        Collector.CollectorConfig.Builder builder = Collector.CollectorConfig.newBuilder()
+        JsonFormat.parser().merge(json, builder)
+        return builder.build()
+    }
+
+    static introspectionQuery(String collectorAddress, String endpoint) {
+        String uri = "http://${collectorAddress}${endpoint}"
+        URL url = new URL(uri)
+        HttpURLConnection connection = null
+
+        try {
+            connection = (HttpURLConnection) url.openConnection()
+
+            // this might be unneeded?
+            connection.setRequestMethod("GET")
+            connection.connect()
+
+            if (connection.responseCode != HttpURLConnection.HTTP_OK) {
+                throw new RuntimeException("IntrospectionQuery failed with ${connection.responseMessage}")
+            }
+            def jsonResponse = connection.inputStream.text
+            return parseJsonToProtobuf(jsonResponse)
+        } catch (Exception e) {
+            throw new RuntimeException("Error making request: ${e.message}", e)
+        } finally {
+            connection.disconnect()
+        }
+    }
+
+    static waitForConfigToHaveState(Kubernetes orchestrator, String state, int timeout = 90, int port = 8080) {
+        def portForwards = orchestrator.createCollectorPortForwards(port)
+
+        log.info "Waiting for Collector Config propagation (${portForwards.size()} pods)"
+        int intervalSeconds = 1
+        int waitTime = 0
+        withRetry(timeout, intervalSeconds) {
+            // if a pod has the right config, remove it from the list
+            // we need to check
+            portForwards.removeAll {
+                def config = introspectionQuery("127.0.0.1:${it.getLocalPort()}", "/state/runtime-config")
+                def configTyped = (Collector.CollectorConfig) config
+                return configTyped.networking.externalIps.enabled.name() == state
+            }
+            assert portForwards.size() == 0
+            waitTime++
```

Comment:
Should  we increase by `intervalSeconds` here? Why not calculating elapsed time

Reply:
Done